$OpenBSD: patch-src_output_OutputCommand_cxx,v 1.1 2015/10/26 16:03:56 dcoppa Exp $

commit ac61d43720393803cb3f6bc5c74aea588e1ca68d
Author: Max Kellermann <max@duempel.org>
Date:   Mon Oct 26 16:29:07 2015 +0100

output/Command: flush the mixer cache when enabling/disabling output

Fixes mixer lag (http://bugs.musicpd.org/view.php?id=4425).

--- src/output/OutputCommand.cxx.orig	Sat Oct 25 00:28:53 2014
+++ src/output/OutputCommand.cxx	Mon Oct 26 16:49:54 2015
@@ -30,6 +30,7 @@
 #include "Internal.hxx"
 #include "PlayerControl.hxx"
 #include "mixer/MixerControl.hxx"
+#include "mixer/Volume.hxx"
 #include "Idle.hxx"
 
 extern unsigned audio_output_state_version;
@@ -47,6 +48,11 @@ audio_output_enable_index(MultipleOutputs &outputs, un
 	ao.enabled = true;
 	idle_add(IDLE_OUTPUT);
 
+	if (ao.mixer != nullptr) {
+		InvalidateHardwareVolume();
+		idle_add(IDLE_MIXER);
+	}
+
 	ao.player_control->UpdateAudio();
 
 	++audio_output_state_version;
@@ -70,6 +76,7 @@ audio_output_disable_index(MultipleOutputs &outputs, u
 	Mixer *mixer = ao.mixer;
 	if (mixer != nullptr) {
 		mixer_close(mixer);
+		InvalidateHardwareVolume();
 		idle_add(IDLE_MIXER);
 	}
 
@@ -94,6 +101,7 @@ audio_output_toggle_index(MultipleOutputs &outputs, un
 		Mixer *mixer = ao.mixer;
 		if (mixer != nullptr) {
 			mixer_close(mixer);
+			InvalidateHardwareVolume();
 			idle_add(IDLE_MIXER);
 		}
 	}
